File: app/src/androidTest/java/br/com/jonathanzanella/myexpenses/bill/ShowBillActivityTest.java
package br.com.jonathanzanella.myexpenses.bill;

import android.content.Intent;
import android.support.test.InstrumentationRegistry;
import android.support.test.filters.MediumTest;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import br.com.jonathanzanella.myexpenses.R;
import br.com.jonathanzanella.myexpenses.database.DatabaseHelper;
import br.com.jonathanzanella.myexpenses.expense.ExpenseRepository;
import br.com.jonathanzanella.myexpenses.helpers.ActivityLifecycleHelper;
import br.com.jonathanzanella.myexpenses.helpers.CurrencyHelper;
import br.com.jonathanzanella.myexpenses.helpers.builder.BillBuilder;

import static android.support.test.InstrumentationRegistry.getInstrumentation;
import static android.support.test.InstrumentationRegistry.getTargetContext;
import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;
import static br.com.jonathanzanella.myexpenses.helpers.TestUtils.waitForIdling;
import static br.com.jonathanzanella.myexpenses.helpers.UIHelper.matchToolbarTitle;

@RunWith(AndroidJUnit4.class)
@MediumTest
public class ShowBillActivityTest {
	@Rule
	public ActivityTestRule<ShowBillActivity> activityTestRule = new ActivityTestRule<>(ShowBillActivity.class, true, false);

	private Bill bill;
	private final ExpenseRepository expenseRepository = new ExpenseRepository();
	private final BillRepository repository = new BillRepository(expenseRepository);

	@Before
	public void setUp() throws Exception {
		new DatabaseHelper(InstrumentationRegistry.getTargetContext()).recreateTables();

		bill = new BillBuilder().build();
		repository.save(bill);
	}

	@After
	public void tearDown() throws Exception {
		ActivityLifecycleHelper.closeAllActivities(getInstrumentation());
	}

	@Test
	public void shows_account_correctly() throws Exception {
		Intent i = new Intent();
		i.putExtra(ShowBillActivity.Companion.getKEY_BILL_UUID(), bill.getUuid());
		activityTestRule.launchActivity(i);

		waitForIdling();

		final String editBillTitle = getTargetContext().getString(R.string.bill) + " " + bill.getName();
		matchToolbarTitle(editBillTitle);

		String balanceAsCurrency = CurrencyHelper.INSTANCE.format(bill.getAmount());
		onView(withId(R.id.act_show_bill_name)).check(matches(withText(bill.getName())));
		onView(withId(R.id.act_show_bill_amount)).check(matches(withText(balanceAsCurrency)));
	}
}

File: app/src/androidTest/java/br/com/jonathanzanella/myexpenses/bill/ShowBillActivityTest.kt
package br.com.jonathanzanella.myexpenses.bill

import android.content.Intent
import android.support.test.InstrumentationRegistry.getInstrumentation
import android.support.test.InstrumentationRegistry.getTargetContext
import android.support.test.espresso.Espresso.onView
import android.support.test.espresso.assertion.ViewAssertions.matches
import android.support.test.espresso.matcher.ViewMatchers.withId
import android.support.test.espresso.matcher.ViewMatchers.withText
import android.support.test.filters.MediumTest
import android.support.test.rule.ActivityTestRule
import android.support.test.runner.AndroidJUnit4
import br.com.jonathanzanella.myexpenses.App
import br.com.jonathanzanella.myexpenses.R
import br.com.jonathanzanella.myexpenses.expense.ExpenseRepository
import br.com.jonathanzanella.myexpenses.helpers.ActivityLifecycleHelper
import br.com.jonathanzanella.myexpenses.helpers.TestUtils.waitForIdling
import br.com.jonathanzanella.myexpenses.helpers.UIHelper.matchToolbarTitle
import br.com.jonathanzanella.myexpenses.helpers.builder.BillBuilder
import br.com.jonathanzanella.myexpenses.helpers.toCurrencyFormatted
import org.junit.After
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@MediumTest
class ShowBillActivityTest {
    @Rule @JvmField
    var activityTestRule = ActivityTestRule(ShowBillActivity::class.java, true, false)

    private var bill: Bill? = null
    private val expenseRepository = ExpenseRepository()
    private val repository = BillRepository(expenseRepository, App.database.billDao())

    @Before
    @Throws(Exception::class)
    fun setUp() {
        App.resetDatabase()

        bill = BillBuilder().build()
        repository.save(bill!!)
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        ActivityLifecycleHelper.closeAllActivities(getInstrumentation())
    }

    @Test
    @Throws(Exception::class)
    fun shows_account_correctly() {
        val i = Intent()
        i.putExtra(ShowBillActivity.KEY_BILL_UUID, bill!!.uuid)
        activityTestRule.launchActivity(i)

        waitForIdling()

        val editBillTitle = getTargetContext().getString(R.string.bill) + " " + bill!!.name
        matchToolbarTitle(editBillTitle)

        val balanceAsCurrency = bill!!.amount.toCurrencyFormatted()
        onView(withId(R.id.act_show_bill_name)).check(matches(withText(bill!!.name)))
        onView(withId(R.id.act_show_bill_amount)).check(matches(withText(balanceAsCurrency)))
    }
}